pkg/stash/stasher.go

```diff
@@ -35,10 +37,16 @@ type stasher struct {
 	s storage.Backend
 }
 
-func (s *stasher) Stash(mod, ver string) error {
+func (s *stasher) Stash(ctx context.Context, mod, ver string) error {
 	const op errors.Op = "stasher.Stash"
-	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*10)
+	_, span := observ.StartSpan(ctx, op.String())
+	defer span.End()
+
+	// create a new context that ditches whatever deadline the caller passed
+	// but keep the tracing info so that we can properly trace the whole thing.
+	ctx, cancel := context.WithTimeout(trace.NewContext(context.Background(), span), time.Minute*10)
```

Comment:
This is awesome 💃 

I would have put this into an observ package so that we can look at it always something like 
`observ.ContextFromSpan(*span) observabilityContext.` but it dis upto you!

Reply:
I figured as much! that is why I think it does not really matter.